packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -464,13 +472,35 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 			uploadP: this.uploadBlob(localId, blob),
 			attached: false,
 			acked: false,
+			abortSignal: signal,
 		};
 		this.pendingBlobs.set(localId, pendingEntry);
 
+		if (signal) {
+			signal.addEventListener(
```

Comment:
we need to clean up the event handler when the create isn't aborted, leaving it hanging leaks memory. you might be able to do something like pendingEntry.handlP.finally(...) to remove it.